### PR TITLE
add meson.project_license() builtin

### LINKED ISSUE #2941: Request to add project_license() method to the meson object
Hi,

In the `meson` builtin object, we have `project_version()` but not `project_license()`, which would be handy for, say, auto-generation of manifest files or similar. It would be useful to add this.

if I get some time, I'll add this myself, but I thought I'd mention it in any case.

### PR DESCRIPTION
**Pull request by martingkelly**
This Fixes #2941.

### CODE REVIEW

File: test cases/common/175 get project license/meson.build [hunk 1]
```diff
@@ -0,0 +1,5 @@
+project('bar', 'c', license: 'Apache')
+
+executable('bar', 'bar.c')
+
```
Comment (snsmac) [inline test cases/common/175 get project license/meson.build]: I would make this an error.

if not (  meson.project_license()[0] == 'Apache' )
    error('The license should be Apache, but it is:  ' + meson.project_license()[0] )
endif

File: test cases/common/175 get project license/meson.build [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+message('This license should be Apache: ' + meson.project_license()[0])
```
Comment (snsmac) [inline test cases/common/175 get project license/meson.build]: I would make this an error.

if not (  meson.project_license()[0] == 'Apache' )
    error('The license should be Apache, but it is:  ' + meson.project_license()[0] )
endif

Comment (martingkelly) [reply test cases/common/175 get project license/meson.build]: Good point, done now.

**Comment by jpakkane** [comment]
Looks good. Since this is new functionality, please add a snippet for the release notes. Thanks.

**Comment by martingkelly** [comment]
Thanks, I rebased and added a release note snippet.

**Pull request merged**